src/swap/saga.ts

```diff
@@ -467,8 +468,36 @@ export function* swapSubmitPreparedSaga(action: PayloadAction<SwapInfoPrepared>)
 
     const feeCurrencyId = getTokenId(networkId, getFeeCurrency(preparedTransactions))
 
-    const swapTxHash = txHashes[txHashes.length - 1]
+    // if there is an approval transaction, add a standby transaction for it
+    if (preparedTransactions.length > 1) {
+      const approvalTx = preparedTransactions[0]
+      const { functionName, args } = yield* call(decodeFunctionData, {
+        abi: erc20.abi,
+        data: approvalTx.data ?? '0x0',
+      })
+      if (functionName === 'approve' && approvalTx.to === fromToken.address && args) {
```

Comment:
Oh nice, this works and isn't too bad.

I was wondering if maybe we would need to include some metadata with prepared transactions so we didn't have to decode further down. But I guess decoding here works too.